crates/sncast/tests/e2e/invoke.rs

```diff
@@ -34,6 +38,80 @@ async fn test_happy_case() {
 
     assert!(matches!(receipt, Invoke(_)));
 }
+#[tokio::test]
+async fn test_tx_reverted() {
+    let class_hash = from_env("CAST_MAP_CLASS_HASH").unwrap();
+    let mut args = default_cli_args();
+
+    args.append(&mut vec![
+        "--account",
+        "user2",
+        "--wait",
+        "invoke",
+        "--contract-address",
+        UDC_ADDRESS,
+        "--function",
+        "deployContract",
+        "--calldata",
+        &class_hash,
+        "0x2 0x01 0x03 0x2 0x1 0x1",
+        "--max-fee",
+        "99999999999999999",
+    ]);
+
+    let snapbox = runner(&args);
+
+    snapbox.assert().stderr_matches(indoc! {r"
+        command: invoke
+        error: Transaction has been reverted
+    "});
+}
+
+#[tokio::test]
+async fn test_tx_rejected() {
+    let provider = create_test_provider();
+    let account = (get_account(
+        "user2",
+        &Utf8PathBuf::from("tests/data/accounts/accounts.json"),
+        &provider,
+        None,
+    )
+    .await)
+        .unwrap();
+
+    let nonce = get_nonce(&provider, "latest", account.address())
+        .await
+        .unwrap();
+    let class_hash = from_env("CAST_MAP_CLASS_HASH").unwrap();
+    let mut args = default_cli_args();
+
+    let nonce_str = format!("{nonce:#02X}",);
+
+    args.append(&mut vec![
+        "--account",
+        "user2",
+        "--wait",
+        "invoke",
+        "--contract-address",
+        UDC_ADDRESS,
+        "--function",
+        "deployContract",
+        "--calldata",
+        &class_hash,
+        "0x2 0x01 0x03 0x2 0x1 0x1",
+        "--max-fee",
+        "99999999999999999",
+        "--nonce",
+        &nonce_str,
+    ]);
+
+    let snapbox = runner(&args);
+
+    snapbox.assert().stderr_matches(indoc! {r"
+        command: invoke
+        error: Transaction has been reverted
```

Comment:
Wasn't the purpose of this test to check if the transaction was rejected? (the name is `test_tx_rejected`)

Reply:
good catch! Tomek🏺